src/Surfnet/Stepup/Identity/Identity.php

```diff
@@ -567,6 +574,12 @@ public function complyWithSecondFactorRevocation(SecondFactorId $secondFactorId,
         }
 
         $vettedSecondFactor->complyWithRevocation($authorityId);
+
+        // Throw an exception if all vetted tokens are cleared
+        // This is used to clear an identity from ra projections
+        if ($this->vettedSecondFactors->isEmpty()) {
+            $this->allSecondFactorsRevoked();
+        }
```

Comment:
Maybe move the if to allSecondFactorsRevoked and rename it to verifyAllSecondFactorsRevoked oid.

Reply:
FYI: Renamed it to `allVettedSecondFactorsRemoved`